pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1160,17 +1183,22 @@ func (p *PriorityQueue) podMatchesEvent(podInfo *framework.QueuedPodInfo, cluste
 
 		// Secondly verify the plugin name matches.
 		// Note that if it doesn't match, we shouldn't continue to search.
-		if evtMatch && intersect(nameSet, podInfo.UnschedulablePlugins) {
+		if evtMatch && intersect(plugins, podInfo.UnschedulablePlugins) {
 			return true
 		}
 	}
 
 	return false
 }
 
-func intersect(x, y sets.Set[string]) bool {
+func intersect(x framework.ClusterEventPlugins, y sets.Set[string]) bool {
 	if len(x) > len(y) {
-		x, y = y, x
+		for v := range y {
```

Comment:
Yes, finding one common entry is enough. That's what the `return true` in the inner loop is for.

Iterating over the smaller set has the advantage that the outer loop finishes sooner. In the most extreme case, y has 100 entries and x zero. When iterating over y, that's 100 loop iterations before returning false.

When iterating over x, that's zero checks.

I had to touch this code because x and y are no longer of the same type, so the trick with swapping no longer works.

Reply:
yeah, thanks for explaining.